firestore/src/common/aggregate_query.cc

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firestore/src/include/firebase/firestore/aggregate_query.h"
+#include "firestore/src/common/cleanup.h"
+#include "firestore/src/common/futures.h"
+#include "firestore/src/common/util.h"
+#include "firestore/src/include/firebase/firestore/aggregate_query_snapshot.h"
+
+#if defined(__ANDROID__)
+// TODO(tomandersen) #include "firestore/src/android/aggregate_query_android.h"
+#else
+#include "firestore/src/main/aggregate_query_main.h"
+#endif  // defined(__ANDROID__)
+
+namespace firebase {
+namespace firestore {
+
+using CleanupFnQuery = CleanupFn<AggregateQuery>;
+
+AggregateQuery::AggregateQuery() {}
+
+AggregateQuery::AggregateQuery(const AggregateQuery& query) {
+  if (query.internal_) {
+    internal_ = new AggregateQueryInternal(*query.internal_);
+  }
+  CleanupFnQuery::Register(this, internal_);
+}
+
+AggregateQuery::AggregateQuery(AggregateQuery&& query) {
+  CleanupFnQuery::Unregister(&query, query.internal_);
+  std::swap(internal_, query.internal_);
+  CleanupFnQuery::Register(this, internal_);
+}
+
+AggregateQuery::AggregateQuery(AggregateQueryInternal* internal)
+    : internal_(internal) {
+  // NOTE: We don't assert internal != nullptr here since internal can be
+  // nullptr when called by the CollectionReference copy constructor.
```

Comment:
This inline comment is not correct because this constructor is _never_ called from the "CollectionReference" copy constructor. Maybe just say that the assertion is not validated because nullptr is an acceptable value for internal_.

Also in `aggregate_query_snapshot.cc`.